api/src/refget/main.py

```diff
@@ -575,10 +624,13 @@ async def sequence(
             raise HTTPException(status_code=500, detail="Internal error. Bad data")
         CACHE[filename] = filehandle
 
-    return StreamingResponse(
-        read_zstd(filehandle, seqstart, seqlength),
-        media_type="text/vnd.ga4gh.refget.v2.0.0+plain; charset=us-ascii",
-    )
+    if len(regions) > 1:
+        content = multi_read_zstd(filehandle, regions)
```

Comment:
I think this  will not stream the result, but instead read it into memory first, then serve it.
To stream, a method (async iterator) must be provided to StreamingResponse, e.g.

if (...):
  return StreamingResponse(read_zstd(), media_type=REFGET...)
else:
    return StreamingResponse(multi_read_zstd(), media_type=REFGET...)